### ISSUE TITLE
WC 3.5.x - fatal error

### ISSUE THREAD

**Issue by whatthefork** [understrap/understrap#961]
Please delete anything that does not apply for you!

## Issue is about..
  - [ ] a Bug Report

## Please tell us about your environment:

  - **WooCommerce 3.5.x 

# Issue Overview for Bugreports

**_Fatal error: Uncaught Error: Call to undefined function wc_review_ratings_enabled() in wp-content/themes/understrap/woocommerce/single-product/rating.php:24_** 

Note that the "wc_review_ratings_enabled" function was apparently added in WC 3.6.x. So calling it in WC 3.5.x throws a fatal error.

**Comment by twiginteractive** [comment]
Looks like a related issue: clean install of WP 5.1.1 (or 5.2) with just WooCommerce 3.6.3 (latest) and Understrap 0.9.3

`Fatal error: Cannot redeclare wc_reviews_enabled() (previously declared in []/html/wp-content/themes/understrap/inc/woocommerce.php:151) in []/wp-content/plugins/woocommerce/includes/wc-conditional-functions.php on line 430`

**Comment by IanDelMar** [comment]
Aren't plugins loaded before the theme?

However, could you try whether this resolves the issue:
In `inc/woocommerce.php` replace
```php
if ( ! function_exists( 'wc_review_ratings_enabled' ) ) {
	/**
	 * Check if reviews are enabled.
	 *
	 * Function introduced in WooCommerce 3.6.0., include it for backward compatibility.
	 *
	 * @return bool
	 */
	function wc_reviews_enabled() {
		return 'yes' === get_option( 'woocommerce_enable_reviews' );
	}
	/**
	 * Check if reviews ratings are enabled.
	 *
	 * Function introduced in WooCommerce 3.6.0., include it for backward compatibility.
	 *
	 * @return bool
	 */
	function wc_review_ratings_enabled() {
		return wc_reviews_enabled() && 'yes' === get_option( 'woocommerce_enable_review_rating' );
	}
}
```
by
```php
if ( ! function_exists( 'understrap_wc_version_check' ) ) {
	function understrap_wc_version_check( $version = '3.6' ) {
		if ( class_exists( 'WooCommerce' ) ) {
			global $woocommerce;
			if ( version_compare( $woocommerce->version, $version, ">=" ) ) {
				return true;
			}
		}
		return false;
	}
}

if ( ! understrap_wc_version_check( '3.6' ) ) {
	/**
	 * Check if reviews are enabled.
	 *
	 * Function introduced in WooCommerce 3.6.0., include it for backward compatibility.
	 * @link https://docs.woocommerce.com/wc-apidocs/function-wc_reviews_enabled.html
	 *
	 * @return bool
	 */
	function wc_reviews_enabled() {
		return 'yes' === get_option( 'woocommerce_enable_reviews' );
	}

	/**
	 * Check if reviews ratings are enabled.
	 *
	 * Function introduced in WooCommerce 3.6.0., include it for backward compatibility.
	 * @link https://docs.woocommerce.com/wc-apidocs/function-wc_review_ratings_enabled.html
	 *
	 * @return bool
	 */
	function wc_review_ratings_enabled() {
		return wc_reviews_enabled() && 'yes' === get_option( 'woocommerce_enable_review_rating' );
	}
}
```

**Comment by twiginteractive** [comment]
@IanDelMar Yep I always thought plugins load before the theme so this error is strange.. unless maybe a hook in WC has a strange/late priority?

I've only tested on latest WC 3.6.3 - not sure if this issue was introduced in a version before or after 3.6

**Comment by jpmorandesign** [comment]
Just updated to WC 3.6.3 and this has not fixed the fatal error.

commenting out the review function does solve it though.

**Comment by IanDelMar** [comment]
I can't reproduce the issue on a clean install of WP 5.2 with WC 3.6.3 Understrap 0.9.3.

**Comment by IanDelMar** [comment]
Ok, I couldn't reproduce it because of the order I was enabling the theme and the plugin.

**Comment by IanDelMar** [comment]
#977 should fix it.

**Comment by Thomas-A-Reinert** [comment]
Has this been ironed out? Would love to close that issue if that´s the case...

**Comment by IanDelMar** [comment]
@Thomas-A-Reinert it has for WC < 3.6, introducing another error for WC 3.6+. The latter is addressed in #977 (tested by @pattonwebz and me) which is waiting to be merged.

**Comment by Thomas-A-Reinert** [comment]
Well, okay, thanks for the update @IanDelMar !
And thanks to @pattonwebz for testing and confirming that your patch works out. Just merged it, so it should work with the latest code.

Closing the issue due to that. Please reopen if needed, k?
